### PR TITLE
Feature/ever present bottom bar

### PR DESCRIPTION
**Pull request by bjornustwo**
New BottomBar

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,9 +24,9 @@
     "react-native-image-marker": "0.1.0",
     "react-native-maps": "0.20.1",
     "react-native-photo-view": "1.5.2",
+    "react-native-remote-svg": "^1.2.0",
```
Comment (vektorustwo) [inline package.json]: Hats off to you sir.

File: src/components/screens/HomeScreen/index.js [hunk 2]
```diff
@@ -48,25 +47,26 @@ type Props = {
...
-      ),
     });
   }
 
+  componentDidMount() {
+    this.props.dispatchShowBottomBar(true);
+  }
+
+  componentWillReceiveProps() {
+    console.log("receive props");
```
Comment (vektorustwo) [inline src/components/screens/HomeScreen/index.js]: deleted

File: src/components/screens/HomeScreen/index.js [hunk 3]
```diff
@@ -128,30 +132,32 @@ class HomeScreen extends Component<Props> {
...
-        />)
-        }
-        renderSectionFooter={({ section }) =>
-          // $FlowFixMe flow doesn't understand me
-          this.renderSectionFooter(section)
-        }
-        keyExtractor={item => item.id}
-        sections={sections}
-      />);
+      <View style={styles.viewContainer}>
```
Comment (vektorustwo) [inline src/components/screens/HomeScreen/index.js]: Delete this View if it isn't needed.

File: src/components/screens/SettingsScreen.js [hunk 4]
```diff
@@ -281,34 +272,38 @@ class SettingsScreen extends Component {
...
       <View style={styles.container}>
-        {this.displayLanguageSegment()}
-        <TouchableOpacity onPress={this.navigateToWelcomeScreen}>
-          <Text style={textStyles.linkText}>{LangService.strings.SEE} {LangService.strings.TUTORIAL}</Text>
-        </TouchableOpacity>
-        <View style={styles.divider} />
-        <TouchableOpacity onPress={this.navigateToDownloadsScreen}>
-          <Text style={textStyles.linkText}>{LangService.strings.OFFLINE_CONTENT}</Text>
-        </TouchableOpacity>
+        <View style={styles.container}>
```
Comment (vektorustwo) [inline src/components/screens/SettingsScreen.js]: Delete this view please.

File: src/components/screens/WebScreen.js [hunk 5]
```diff
@@ -1,24 +1,31 @@
...
 });
 
 const WebScreen = (props) => {
   const { url } = props.navigation.state.params;
 
   return (
-    <ViewContainer>
-      <WebView source={{ uri: url }} />
-    </ViewContainer>
+    <View style={styles.container}>
```
Comment (vektorustwo) [inline src/components/screens/WebScreen.js]: Dubbeltrubbel.

File: src/components/screens/WebScreen.js [hunk 6]
```diff
@@ -1,24 +1,31 @@
 import React from "react";
-import { WebView, StyleSheet } from "react-native";
+import { WebView, View, StyleSheet } from "react-native";
 import PropTypes from "prop-types";
-import ViewContainer from "../shared/view_container";
 import {
   Colors,
 } from "../../styles/";
+import BottomBarView from "../shared/BottomBarView";
```
Comment (vektorustwo) [inline src/components/screens/WebScreen.js]: Bort!

File: src/components/shared/BottomBarIcon/index.js [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
... 21.699219 9.398438 21.800781 9.101563 21.601563 8.800781 L 19.601563 5.300781 C 19.5 5 19.199219 4.898438 19 5 L 16.601563 6.101563 C 16.101563 5.699219 15.398438 5.300781 14.800781 5.101563 L 14.5 2.5 C 14.5 2.199219 14.300781 2 14 2 L 10 2 C 9.699219 2 9.5 2.199219 9.5 2.5 L 9.199219 5 C 8.5 5.300781 7.898438 5.601563 7.398438 6 L 5 5 C 4.800781 4.898438 4.5 5 4.398438 5.199219 L 2.398438 8.699219 C 2.199219 9 2.199219 9.300781 2.5 9.398438 L 4.601563 11 C 4.601563 11.300781 4.5 11.699219 4.5 12 C 4.5 12.300781 4.5 12.699219 4.601563 13 L 2.5 14.5 C 2.300781 14.601563 2.199219 14.898438 2.398438 15.199219 L 4.398438 18.699219 C 4.5 19 4.800781 19.101563 5 19 L 7.398438 17.898438 C 7.898438 18.300781 8.601563 18.699219 9.199219 18.898438 L 9.5 21.5 C 9.5 21.800781 9.699219 22 10 22 L 14 22 C 14.300781 22 14.5 21.800781 14.5 21.5 L 14.800781 18.898438 C 15.5 18.601563 16.101563 18.300781 16.601563 17.898438 L 19 19 C 19.199219 19.101563 19.5 19 19.601563 18.800781 L 21.601563 15.300781 C 21.699219 15.101563 21.699219 14.800781 21.5 14.601563 Z M 12 16 C 9.800781 16 8 14.199219 8 12 C 8 9.800781 9.800781 8 12 8 C 14.199219 8 16 9.800781 16 12 C 16 14.199219 14.199219 16 12 16 Z "/>
```
Comment (vektorustwo) [inline src/components/shared/BottomBarIcon/index.js]: Tjusig! 🌟

File: src/components/shared/BottomBarIcon/index.js [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+          source={getIcon(this.props.index, color)}
+          style={styles.icon}
+        />
+        <Text style={[styles.text, { color }]}>{getText(this.props.index)}</Text>
+      </TouchableOpacity>
+    );
+    return bottomBarIcon;
+  }
+
+  onTouchIcon(index: number, selected: boolean) {
```
Comment (vektorustwo) [inline src/components/shared/BottomBarIcon/index.js]: You can add a private flow type enum for this, if you want.

File: src/components/shared/BottomBarView/style.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
+import { StyleSheet } from "react-native";
+import { Colors } from "../../../styles";
+
+export default StyleSheet.create({
+  viewContainer: {
+    flex: 0.11,
```
Comment (vektorustwo) [inline src/components/shared/BottomBarView/style.js]: Would prefer fixed height for this component.

**Comment by vektorustwo** [comment]
Visual artefacts on iPhone 5S.

simulator screen shot - iphone 5s - 2018-06-07 at 14 41 01

**Comment by bjornustwo** [comment]
For some reason Travis won't accept this PR anymore. Not sure why

**Pull request merged**